src/aap_eda/settings/default.py

```diff
@@ -636,6 +636,39 @@ def get_rulebook_process_log_level() -> RulebookProcessLogLevel:
 # Organization and object roles will come from create_initial_data
 ANSIBLE_BASE_ROLE_PRECREATE = {}
 
+# --------------------------------------------------------
+# ANSIBLE BASE RESOURCE API CLIENT
+# --------------------------------------------------------
+RESOURCE_SERVER = {
+    "URL": settings.get("RESOURCE_SERVER__URL", "https://localhost"),
+    "SECRET_KEY": settings.get("RESOURCE_SERVER__SECRET_KEY", ""),
+    "VALIDATE_HTTPS": settings.get("RESOURCE_SERVER__VALIDATE_HTTPS", False),
+}
+RESOURCE_JWT_USER_ID = settings.get("RESOURCE_JWT_USER_ID", "")
+RESOURCE_SERVICE_PATH = settings.get(
+    "RESOURCE_SERVICE_PATH", "/api/gateway/v1/service-index/"
+)
+
+if DIRECT_SHARED_RESOURCE_MANAGEMENT_ENABLED:
+    if (
+        RESOURCE_SERVER["URL"]
+        and RESOURCE_SERVER["SECRET_KEY"]
+        and RESOURCE_JWT_USER_ID
+    ):
+        RQ_PERIODIC_JOBS.append(
+            {
+                "func": "aap_eda.tasks.shared_resources.resync_shared_resources",  # noqa E501
+                "interval": 900,
+                "id": "resync_shared_resources",
+            }
+        )
+    else:
+        raise ImproperlyConfigured(
```

Comment:
Should we raise this error in another PR? What happens if the installer or operator has not set these ENVS?

Reply:
It's better to catch it as early as possible otherwise imagine the amount of time spent trying to debug this. They will have the whole env setup and then the syncs won't happen. I don't think this is an optional feature if that field has been setup the others have to be set too.

Reply:
I agree with @mkanoor 
What we can do is merge this PR in sync with our mates @jamesmarshall24 @dsavineau and @rooftopcellist to minimize the window of errors.